src/integer_object.cpp

```diff
@@ -226,11 +226,13 @@ Value IntegerObject::pow(Env *env, Value arg) const {
 }
 
 Value IntegerObject::cmp(Env *env, Value arg) {
-    if (!arg.is_fast_integer()) {
-        arg.unguard();
-        if (!arg->is_integer() && !arg->is_float())
-            return NilObject::the();
-    }
+    // Check if we might want to coerce the value
+    if (!arg.is_fast_integer() && !arg->is_integer() && !arg->is_float())
+        arg = Natalie::coerce(env, arg, this, false).second;
+
+    // Check if compareable
+    if (!arg.is_fast_integer() && !arg->is_integer() && !arg->is_float())
+        return NilObject::the();
```

Comment:
Sadly I cannot think of a technique to eliminate this code duplication that itself would not make the code harder to read. 😞 I considered a lambda, but I don't think the added complexity is worth eliminating one duplicated line. 🤷

Reply:
Yeah I thought about a lambda as well... But what do you think about a short method? Something like `is_comparable_with`?

Reply:
I think a lambda would bring less overhead and the compiler is very likely to inline it, so I don't think it'd be too bad :)

Reply:
I added a lambda for it! @seven1m If you think it's too complex you can just remove the last commit. I'll be away for the next few days so I just did that in advance 😄 🎄